### ISSUE TITLE
NoSuperfluousPhpdocTagsFixer mis-identifies parameter name

### ISSUE THREAD

**Issue by AurelC2G** [FriendsOfPHP/PHP-CS-Fixer#3976]
Hi,

In `NoSuperfluousPhpdocTagsFixer`, there seems to be an issue with the regex on line 111 [1]: if the description of a parameter contains a variable name, the fixer will assume that the `@param` annotation applies to the name from the description instead of the actual param name. If that name from the description does not match any of the arguments from the function, the annotation gets wrongly removed (see example below).

[1] https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/2.12/src/Fixer/Phpdoc/NoSuperfluousPhpdocTagsFixer.php#L111

#### The PHP version you are using (`$ php -v`):
PHP 7.2.7-0ubuntu0.18.04.2

#### PHP CS Fixer version you are using (`$ php-cs-fixer -V`):
PHP CS Fixer 2.12.2 Long Journey

#### The command you use to run PHP CS Fixer:
`php-cs-fixer fix`

#### The configuration file you are using, if any:
```php
<?php
$finder = PhpCsFixer\Finder::create()
    ->in(__DIR__.'/src')
    ->ignoreDotFiles(true)
    ->ignoreVCS(true)
    ->files()
    ->name('*.php')
;

return PhpCsFixer\Config::create()
    ->setFinder($finder)
    ->setRules([
        'no_superfluous_phpdoc_tags' => true,
    ])
;
```

#### If applicable, please provide minimum samples of PHP code (as plain text, not screenshots):
 * before running PHP CS Fixer (no changes):
```php
/**
 * Function summary here
 *
 * @param $value Some description that includes a $variable
 */
function f($value)
{
}
```

 * with unexpected changes applied when running PHP CS Fixer:
```php
/**
 * Function summary here
 *
 */
function f($value)
{
}
```

 * with the changes you expected instead:
The `@param` annotation should not get removed.

**Comment by SpacePossum** [comment]
thanks for reporting @AurelC2G :+1:

**Comment by julienfalque** [comment]
Should be fixed by #3977.
